contracts/dtoken/src/user_flow_protection.rs

```diff
@@ -0,0 +1,91 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::{AccountId, BlockHeight, log, near_bindgen};
+use near_sdk::collections::{UnorderedMap};
+use near_sdk::env::block_height;
+
+const BLOCKS_TO_NEXT_OPERATION: BlockHeight = 100;
+
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct UserPerform {
+    last_user_perform: UnorderedMap<AccountId, BlockHeight>,
+}
+
+impl Default for UserPerform {
+    fn default() -> Self {
+        Self {
+            last_user_perform: UnorderedMap::new(b"s".to_vec()),
+        }
+    }
+}
+
+#[near_bindgen]
+impl UserPerform {
+    pub fn set_last_user_perform(&mut self, account_id: AccountId) {
+        let block_height = block_height();
```

Comment:
From my perspective, this parameter should be in the signature of the setter method.